src/Illuminate/Console/Scheduling/ScheduleBackgroundCommand.php

```diff
@@ -54,8 +54,24 @@ public function __construct(Schedule $schedule)
      */
     public function handle()
     {
-        collect($this->schedule->events())->filter(function ($value) {
-            return $value->mutexName() == $this->argument('id');
-        })->each->callAfterCallbacks($this->laravel);
+        if (! $event = $this->findEventMutex()) {
+            $this->error('No scheduled event could be found that matches the given id.');
+
+            return;
+        }
+
+        $event->runCommandInForeground($this->laravel);
+    }
+
+    /**
+     * Find the event that matches the id.
+     *
+     * @return mixd
+     */
+    protected function findEventMutex()
+    {
+        return collect($this->schedule->events())->filter(function ($value) {
```

Comment:
Actually here you filter your collection, then you return the first item of this filtered collection.
With Arr::first, you return the first item that match the condition in the closure.

This will achieve the same thing, but much faster:
```php
return Arr::first($this->schedule->events(), (function (Event $value) {
    return $value->mutexName() === $this->argument('id');
});
```